frontend/packages/console-app/src/components/cloud-shell/CloudShellExec.tsx

```diff
@@ -74,13 +76,27 @@ const CloudShellExec: React.FC<CloudShellExecProps> = ({
 
   const tick = useActivityTick(workspaceName, namespace);
 
-  const onData = React.useCallback(
-    (data: string): void => {
-      tick();
-      ws.current?.send(`0${Base64.encode(data)}`);
-    },
-    [tick],
-  );
+  React.useEffect(() => {
+    let startTime;
+    let tickReq;
+    const handleTick = (timestamp) => {
+      if ((!startTime || timestamp - startTime >= TICK_INTERVAL) && isActiveTab) {
+        startTime = timestamp;
+        tick();
+      }
+      tickReq = window.requestAnimationFrame(handleTick);
+    };
+
+    tickReq = window.requestAnimationFrame(handleTick);
+
+    return () => {
+      window.cancelAnimationFrame(tickReq);
+    };
+  }, [isActiveTab, tick]);
```

Comment:
Every tab will be running this `requestAnimationFrame` loop at all times even if it is not the active tab. You could alleviate this by not requesting a frame unless it is the active tab. 

That being said, the `tick` isn't specific to a tab and could be moved higher in the component tree providing we know higher up when we have a connection.

Reply:
@christianvogt Let us know how to proceed?
1. We could have this call higher up at the `MultiTabTerminal` level.  But the problem is the `activityTick('workspacename', 'namespace')`. This `workspaceName`, `namespace' data is only available at an individual terminal Level. To have this data higher up would require a lot of refactoring.

2. assign tickReq only for the activeTab else assign null. something like `tickReq = isActive ?  window.requestAnimationFrame(handleTick) : null ` 
WDYT?
cc: @sahil143 @invincibleJai

Reply:
We only want to send the tick if there's an open connection. It's either each terminal is responsible for sending the tick when active. Or each terminal is responsible for issuing a callback of its status to the parent which can then control sending the tick with the information provided by its terminal children. Having a single "manager" that is responsible for the tick would be preferred but both options work.

I didn't look if there's any performance implications of the current implementation. I assumed that each terminal would be running a tight animation loop, therefore doing _some_ work at all times; but it's not a heavy load.